src/app/content/components/TableOfContents/keyboardSupport.hook.tsx

```diff
@@ -0,0 +1,171 @@
+import { HTMLElement, Element } from '@openstax/types/lib.dom';
+import { LinkedArchiveTreeNode } from '../../types';
+import React from 'react';
+
+export interface KeyboardSupportProps {
+  event: React.KeyboardEvent<HTMLAnchorElement>;
+  item: LinkedArchiveTreeNode;
+  isOpen?: boolean;
+  onSelect: () => void;
+}
+
+const getTreeItems = () => {
+  const treeItems = Array.from(document?.querySelectorAll('a[role="treeitem"]') as ArrayLike<HTMLElement>);
+  const treeItemResults = treeItems.filter((el) => {
+    const style = window?.getComputedStyle(el);
+    return (style?.display !== 'none');
+  });
+
+  return treeItemResults;
+};
+
+const focusNextTreeItem = (filteredTreeItems: Element[], currentItemIndex: number) => {
+  document?.querySelector<HTMLElement>(`[id="${filteredTreeItems[currentItemIndex + 1]?.id}"]`)?.focus();
+};
+
+const focusPrevTreeItem = (filteredTreeItems: Element[], currentItemIndex: number) => {
+  document?.querySelector<HTMLElement>(`[id="${filteredTreeItems[currentItemIndex - 1]?.id}"]`)?.focus();
+};
+
+const focusParent = (filteredTreeItems: Element[], currentItemIndex: number) => {
+  (document?.querySelector(
+    `[id="${filteredTreeItems[currentItemIndex]?.id}"]`)
+    ?.parentElement?.parentElement
+    ?.previousElementSibling as HTMLElement)
+    ?.focus();
+};
+
+const focusFirstTreeItem = (filteredTreeItems: Element[]) => {
+  document?.querySelector<HTMLElement>(`[id="${filteredTreeItems[0]?.id}"]`)?.focus();
+};
+
+const focusLastTreeItem = (filteredTreeItems: Element[]) => {
+  document?.querySelector<HTMLElement>(`[id="${filteredTreeItems[filteredTreeItems.length - 1]?.id}"]`)?.focus();
+};
+
+const focusOnTab = (filteredTreeItems: Element[], shiftKey: boolean) => {
+  const pageElements = Array.from(
+    document?.querySelectorAll('input, button, select, textarea, a[href]') as ArrayLike<HTMLElement>
+  );
+  let index = pageElements.indexOf(document?.activeElement as HTMLElement);
+  if (shiftKey) {
+    focusFirstTreeItem(filteredTreeItems);
```

Comment:
why are you changing the treeitem focus and then focusing something else right after?